### PR TITLE
Prebuilt uefi installer script

### PR DESCRIPTION
**Pull request by ric96**

### CODE REVIEW

File: install-prebuilt-uefi/hikey-efi-prebuilt.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+#set -e
+
+if [ "$#" -ne 1 ]; then
+	echo "Usage: hikey-efi-flash-image <release/debug>"
+	exit 1
+fi
+
+RELEASE=$1
+UEFI_DIR=${1:-./uefi}
+UEFI_URL="http://builds.96boards.org/snapshots/reference-platform/components/uefi-staging/14/hikey960/$RELEASE/"
```
Comment (docularxu) [inline install-prebuilt-uefi/hikey-efi-prebuilt.sh]: Your scripts uses build #14. I know so far this is the only working one. You can use that as default. But I do want you give the user freedom to choose. See, I would like something:
UEFI_URL=${BASE_URL}/${VERSION}/${PRODUCT}/${RELEASE}

File: install-prebuilt-uefi/hikey-efi-prebuilt.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
```
Comment (docularxu) [inline install-prebuilt-uefi/hikey-efi-prebuilt.sh]: This works for hikey960 only. It doesn't support original hikey (hi6220). So please change the file name to 'hikey960-efi-prebuilt.sh'

File: install-prebuilt-uefi/hikey-efi-prebuilt.sh (hunk 1 above)
Comment (docularxu) [reply install-prebuilt-uefi/hikey-efi-prebuilt.sh]: Here is a script I wrote locally. Would you please incorporate it into yours?

BASE_URL=http://builds.96boards.org/snapshots/reference-platform/components/uefi-staging
VERSION=20
PRODUCT=hikey960
RELEASE=release

while [[ $# -gt 0 ]]
do
key="$1"

case $key in
	-h)
		echo "Look the file to get help"
		echo "Use -d or -r to download debug or release version"
		echo "Use -v to specify build no."
		echo "Example: "$0" -r -v 20"
		echo "    , this downloads build #20 release version"
		exit 0
		;;
	-d|--debug)
		RELEASE=debug
		;;
	-r|--release)
		RELEASE=release
		;;
	-v|--version)
		VERSION=$2
		shift
		;;
	*)
		echo "Unknown option. Use -h for help"
		exit 0
		;;
esac
shift
done

TARGET_URL=${BASE_URL}/${VERSION}/${PRODUCT}/${RELEASE}

**Comment by docularxu** [comment]
Thanks for submitting this to us. Actually we/Linaro LT are preparing to move from obsolete fastboot bootloader to the latest UEFI. There are two steps involved:
1. Need to prepare a script and document to guide users to smoothly switch from old bootloader to UEFI.
2. Need a script and document about how to update UEFI.

Your commit is useful for Step 2. But before that, we need to close Step 1. In summary, I will cherry-pick the readme.md file you submitted, but I will not merge your other commits until I settle down Step 1. Action is on me.

**Comment by ric96** [comment]
@docularxu Hi, I've made the requested modifications

**Comment by hzhuang1** [comment]
@ric96 Thanks a lot for your patches. There're a few revert patches. Could you clean the revert patches first?

**Comment by ric96** [comment]
@hzhuang1 Sorry for the delay, It should be clean now

**Pull request merged**